src/main/handlers/audioExtract.ts

```diff
@@ -1,22 +1,29 @@
 import path from 'path';
 import { path as ffmpegPath } from '@ffmpeg-installer/ffmpeg';
 import ffmpeg from 'fluent-ffmpeg';
+import { Project } from 'sharedTypes';
+import { audioExtractStoragePath, mkdir } from '../util';
 
 ffmpeg.setFfmpegPath(ffmpegPath);
 
-export default function extractAudio(
-  absolutePathToMediaFile: string
-): Promise<string> {
+export default function extractAudio(project: Project): Promise<string> {
+  if (project.mediaFilePath === null || project.id === null) {
+    return new Promise((_resolve, reject) => {
+      const errorMessage = 'Project mediaFilePath or ID cannot be null';
+      reject(errorMessage);
+    });
+  }
+
+  mkdir(audioExtractStoragePath());
+
   const pathToSaveMedia = path.join(
-    process.cwd(),
-    'assets',
-    'audio',
-    'audio.wav'
+    audioExtractStoragePath(),
```

Comment:
This change will break the hardcoded file path in transcriptionHandler.ts (line 70) which it uses to grab the audio file. We need to change the transcriptionHandler so the path is passed in or something

Reply:
Nice catch, I wasnt able to spot this since i always had the audio.wav file! Here is the commit where I changed the method: https://github.com/patrickbrett/mlvet/pull/63/commits/4129bdb2e424bec62d06ec0a42db625973e0df24

You can find the rest of the commits which update the variable names and datatypes within the IPC related files in the next commits below.

Reply:
Fantastic, that's a much better solution all round. Great work